runtime/compiler/runtime/J9CodeCacheManager.cpp

```diff
@@ -298,6 +300,19 @@ J9::CodeCacheManager::allocateCodeCacheSegment(size_t segmentSize,
       {
       vmemParams.pageSize = largeCodePageSize;
       vmemParams.pageFlags = config.largeCodePageFlags();
+
+      // Recalculate the page size if using large pages
+      UDATA *pageSizes = j9vmem_supported_page_sizes();
+      if(vmemParams.pageSize > ccTotalSizeB)
+         {
+         for (UDATA pageIndex = 0; 0 != pageSizes[pageIndex]; ++pageIndex)
+            {
+            if(pageSizes[pageIndex] <= ccTotalSizeB)
+               vmemParams.pageSize = pageSizes[pageIndex];
+            }
+         if (config.verboseCodeCache())
```

Comment:
We need a trace point for this rare condition, because vlogs are not always enabled.

Reply:
@mpirvu I added the tracepoint but I am unable to get it printed in my local testing. Please can you see if there are any obvious issues with the trace code. After the build `ut_j9jit.h` contains:
```
#if UT_TRACE_OVERHEAD >= 1
#define Trc_JIT_CodeCacheLargePageSizeFailure(P1, P2) do { /* tracepoint name: j9jit.93 */ \
        if ((unsigned char) j9jit_UtActive[93] != 0){ \
                j9jit_UtModuleInfo.intf->Trace((void *)NULL, &j9jit_UtModuleInfo, ((93u << 8) | j9jit_UtActive[93]), "\6\6", P1, P2);} \
        } while(0)
#else
#define Trc_JIT_CodeCacheLargePageSizeFailure(P1, P2)   /* tracepoint name: j9jit.93 */
#endif
```

The verbose logs prints:
`#CODECACHE:  Warning: Using page size 4096 instead of large page size 2097152`

However `-Xtrace:print=j9jit.90-93 ` doesn't print the trace point. Do you have any ideas?

Reply:
Do you see other trace points?

Reply:
I added a printf and I see this:
```
allocateCodeCacheSegment
22:26:38.543*0x1aa00           j9jit.0        - Trace engine initialized for module j9jit
```
As you can see the allocation for the code cache happens before the trace engine initialization, thus your trace point will not appear.

Reply:
We have to eliminate the trace point in this case. We should however print the verbose message for verbose={compilePerformance} too.

Reply:
ok.. I will add verbose performance too. Thanks for pointing out that the trace engine initialization happens after and hence trace points were not getting captured.